test/controllers/image_shares_controller_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+class ImageSharesControllerTest < ActionDispatch::IntegrationTest
+  def test_should_get_new
+    image = Image.create!(imageurl: 'http://abc.png')
+    get new_image_image_share_path(image)
+    assert_response :ok
+  end
+
+  def test_should_create_image_shares_email
+    image = Image.create!(imageurl: 'http://abc.png')
+    share_params = { share: { email: 'a@b.com' } }
+
+    post image_image_shares_path(image), params: { image: share_params }
+
+    assert_redirected_to images_path
+    assert_equal 'You have successfully shared an image.', flash[:success]
+  end
+
+  def test_should_fail_to_create_image_shares_with_invalid_email
```

Comment:
Can we write a few more tests to cover the regex?

Reply:
Yes, we'll pull these into a form class as suggested and test common validation cases.